packages/net/tcp_connection.pony

```diff
@@ -478,65 +513,57 @@ actor TCPConnection
     This occurs only with IOCP on Windows.
     """
     ifdef windows then
-      var rem = len.usize()
-
-      if rem == 0 then
+      if len == 0 then
         // IOCP reported a failed write on this chunk. Non-graceful shutdown.
-        try _pending.shift() end
         _hard_close()
         return
       end
 
-      while rem > 0 do
-        try
-          let node = _pending.head()
-          (let data, let offset) = node()
-          let total = rem + offset
-
-          if total < data.size() then
-            node() = (data, total)
-            rem = 0
-          else
-            _pending.shift()
-            rem = total - data.size()
-          end
-        end
+      try
+        _manage_pending_buffer(len.usize(), _pending_writev_total, _pending.size())
       end
 
-      if _pending.size() < 16 then
+      if _pending_sent < 16 then
         // If fewer than 16 asynchronous writes are scheduled, remove
         // backpressure. The choice of 16 is rather arbitrary and probably
         // needs to be tuned.
         _release_backpressure()
       end
     end
 
-  fun ref _pending_writes() =>
+  fun ref _pending_writes(): Bool =>
     """
     Send pending data. If any data can't be sent, keep it and mark as not
-    writeable. On an error, dispose of the connection.
+    writeable. On an error, dispose of the connection. Returns whether
+    it sent all pending data or not.
     """
     ifdef not windows then
-      while _writeable and (_pending.size() > 0) do
+      // TODO: Make writev_batch_size user configurable
+      let writev_batch_size: USize = @pony_os_writev_max[I32]().usize()
+      var num_to_send: USize = 0
+      var bytes_to_send: USize = 0
+      while _writeable and (_pending_writev_total > 0) do
         try
-          let node = _pending.head()
-          (let data, let offset) = node()
+          //determine number of bytes and buffers to send
+          if (_pending_writev.size()/2) < writev_batch_size then
+            num_to_send = _pending_writev.size()/2
```

Comment:
Can you put a single space between the `/` operator and its operands, on both this line and the line above it?